service/worker/workerdeployment/replaytester/worker/worker.go

```diff
@@ -184,23 +193,52 @@ func main() {
 
 //nolint:revive
 func verifyDeployment(dHandle client.WorkerDeploymentHandle,
-	expectedCurrentVersion string,
-	expectedRampingVersion string,
+	expectedCurrentVersionBuildId string,
+	expectedRampingVersionBuildId string,
+	expectedRampPercentage float32,
 	expectedDrainageStatus client.WorkerDeploymentVersionDrainageStatus,
 ) {
 	describeResponse, err := dHandle.Describe(context.Background(), client.WorkerDeploymentDescribeOptions{})
 	if err != nil {
 		log.Fatalln("Unable to describe deployment", err)
 	}
-	if describeResponse.Info.RoutingConfig.CurrentVersion != expectedCurrentVersion {
-		log.Fatalln("Current version is not ", expectedCurrentVersion)
+	if cv := describeResponse.Info.RoutingConfig.CurrentVersion; cv != nil {
+		if cv.BuildId != expectedCurrentVersionBuildId {
+			log.Fatalln(fmt.Sprintf("Current version build id is %s not %s", cv.BuildId, expectedCurrentVersionBuildId))
+		}
+	} else {
+		if expectedCurrentVersionBuildId != "" {
+			log.Fatalln("Current version is empty, expected build id ", expectedCurrentVersionBuildId)
+		}
+	}
+
+	if rv := describeResponse.Info.RoutingConfig.RampingVersion; rv != nil {
+		if rv.BuildId != expectedRampingVersionBuildId {
+			log.Fatalln(fmt.Sprintf("Ramping version build id is %s not %s", rv.BuildId, expectedRampingVersionBuildId))
+		}
+	} else {
+		if expectedRampingVersionBuildId != "" {
+			log.Fatalln("Ramping version is empty, expected build id ", expectedRampingVersionBuildId)
+		}
+	}
+
+	if rp := describeResponse.Info.RoutingConfig.RampingVersionPercentage; rp != expectedRampPercentage {
```

Comment:
improved error messages here only because I ran into some dumb errors and wanted better messages